src/algebra/algebra/subalgebra.lean

```diff
@@ -34,29 +34,30 @@ variables {R : Type u} {A : Type v} {B : Type w}
 variables [comm_semiring R] [semiring A] [algebra R A] [semiring B] [algebra R B]
 include R
 
-instance : has_coe (subalgebra R A) (subsemiring A) :=
-⟨λ S, { ..S }⟩
+instance : has_coe_t (subalgebra R A) (set A) := ⟨λ s, s.carrier⟩
+instance : has_mem A (subalgebra R A) := ⟨λ x p, x ∈ (p : set A)⟩
+instance : has_coe_to_sort (subalgebra R A) := ⟨_, λ p, {x : A // x ∈ p}⟩
```

Comment:
Can we add at comment here, explaining why we don't want to define coercions to submodule and subsemiring?

Reply:
The main reason is just "none of the other objects do it, and it creates more API that we have to write"